features/lorawan/lorastack/phy/LoRaPHYAU915.h

```diff
@@ -57,317 +57,54 @@ class LoRaPHYAU915 : public LoRaPHY{
-     *
-     * \retval The status of the operation, according to the LoRaMAC specification.
-     *         Returns -1, if the functionality is not implemented. In this case, the end node
-     *         shall ignore the command.
-     */
-    virtual int8_t setup_tx_params(TxParamSetupReqParams_t* txParamSetupReq );
-
-    /*!
-     * \brief The function processes a DlChannel request.
-     *
-     * \param [in] dlChannelReq A pointer to the function parameters.
-     *
-     * \retval The status of the operation, according to the LoRaMAC specification.
-     */
-    virtual uint8_t dl_channel_request(DlChannelReqParams_t* dlChannelReq );
-
-    /*!
-     * \brief Alternates the datarate of the channel for the join request.
-     *
-     * \param [in] alternateDr A pointer to the function parameters.
-     *
-     * \retval The datarate to apply.
-     */
-    virtual int8_t get_alternate_DR(AlternateDrParams_t* alternateDr );
+                                     uint8_t* nbBytesParsed);
 
-    /*!
-     * \brief Calculates the back-off time.
-     *
-     * \param [in] calcBackOff A pointer to the function parameters.
-     */
-    virtual void calculate_backoff(CalcBackOffParams_t* calcBackOff );
+    virtual uint8_t accept_rx_param_setup_req(rx_param_setup_req_t* params);
 
-    /*!
-     * \brief Searches and sets the next random available channel.
-     *
-     * \param [in]  nextChanParams The parameters for the next channel.
-     *
-     * \param [out] channel The next channel to use for TX.
-     *
-     * \param [out] time The time to wait for the next transmission according to the duty
-     *              cycle.
-     *
-     * \param [out] aggregatedTimeOff Updates the aggregated time off.
-     *
-     * \retval The function status [1: OK, 0: Unable to find a channel on the current datarate].
-     */
-    virtual bool set_next_channel(NextChanParams_t* nextChanParams,
-                                   uint8_t* channel, lorawan_time_t* time,
-                                   lorawan_time_t* aggregatedTimeOff );
+    virtual int8_t get_alternate_DR(uint8_t nb_trials);
 
-    /*!
-     * \brief Adds a channel.
-     *
-     * \param [in] channelAdd A pointer to the function parameters.
-     *
-     * \retval The status of the operation.
-     */
-    virtual lorawan_status_t add_channel(ChannelAddParams_t* channelAdd );
+    virtual bool set_next_channel(channel_selection_params_t* next_chan_params,
+                                  uint8_t* channel, lorawan_time_t* time,
+                                  lorawan_time_t* aggregate_timeoff);
 
-    /*!
-     * \brief Removes a channel.
-     *
-     * \param [in] channelRemove A pointer to the function parameters.
-     *
-     * \retval True, if the channel was removed successfully.
-     */
-    virtual bool remove_channel(ChannelRemoveParams_t* channelRemove );
+    virtual void set_tx_cont_mode(cw_mode_params_t* params);
```

Comment:
LoRaPHY::set_tx_cont_mode has also a default parameter frequency. It must be defined in inherited classes too (this does not even compile with ARM compilers).
Same applies for other regions (KR920, US915 and US915Hybrid) too where this method has been overridden.